pkg/credentialprovider/plugin/plugin.go

```diff
@@ -447,6 +453,22 @@ func (p *pluginProvider) provide(image, podNamespace, podName string, podUID typ
 		return credentialprovider.DockerConfig{}
 	}
 
+	if len(serviceAccountToken) == 0 && len(response.ServiceAccountTokenCacheType) > 0 {
+		klog.Errorf("credential provider plugin not allowed to return serviceAccountTokenCacheType when serviceAccountToken is not provided")
+		return credentialprovider.DockerConfig{}
+	}
+	if len(serviceAccountToken) > 0 {
```

Comment:
also sanity check if they echoed the token back verbatim as a Password entry and require the ServiceAccountTokenCacheType to be `Token` in that case